File: boj/c++/10845.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <string>

using namespace std;

int arr[10002];

int main() {
    int head = 1;
    int tail = 1;
    int N;

    cin >> N;

    for (int i = 1; i <= N; i++) {
        string command;
        cin >> command;
        if (command == "push") {
            int input;
            cin >> input;
            arr[tail] = input;
            tail++;
        } else if (command == "pop") {
            if (head != tail) {
                cout << arr[head] << '\n';
                head++;
            } else {
                cout << -1 << '\n';
            }
        } else if (command == "size") {
            cout << tail - head << '\n';
        } else if (command == "empty") {
            if (tail - head) {
                cout << 0 << '\n';
            } else {
                cout << 1 << '\n';
            }
        } else if (command == "front") {
            if (head != tail) {
                cout << arr[head] << '\n';
            } else {
                cout << -1 << '\n';
            }
        } else if (command == "back") {
            if (head != tail) {
                cout << arr[tail - 1] << '\n';
            } else {
                cout << -1 << '\n';
            }
        } else {
            cout << "Error\n";
        }
    }

    return 0;
}

File: boj/python/10845.py
import sys
from collections import deque

N = int(sys.stdin.readline().rstrip())
s = deque()
for _ in range(N):
    cmds = sys.stdin.readline().rstrip().split()
    if len(cmds) > 1:
        s.append(int(cmds[1]))
    elif cmds[0] == "front":
        if len(s) > 0:
            print(s[0])
        else:
            print(-1)
    elif cmds[0] == "back":
        if len(s) > 0:
            print(s[-1])
        else:
            print(-1)
    elif cmds[0] == "pop":
        if len(s) > 0:
            print(s.popleft())
        else:
            print(-1)
    elif cmds[0] == "size":
        print(len(s))
    elif cmds[0] == "empty":
        print(1) if len(s) == 0 else print(0)
